target_chains/cosmwasm/tools/src/chains-manager/chain-executor.ts

```diff
@@ -0,0 +1,115 @@
+import { CosmwasmExecutor } from "./cosmwasm";
+import { InjectiveExecutor } from "./injective";
+
+export type Fund = {
+  denom: string;
+  amount: string;
+};
+export type StoreCodeRequest = {
+  contractBytes: Buffer;
+};
+export type StoreCodeResponse = {
+  codeId: number;
+  txHash: string;
+};
+
+export type InstantiateContractRequest = {
+  codeId: number;
+  instMsg: object;
+  label: string;
+};
+export type InstantiateContractResponse = {
+  contractAddr: string;
+  txHash: string;
+};
+
+export type ExecuteContractRequest = {
+  contractAddr: string;
+  msg: object;
+  funds?: Fund[];
+};
+export type ExecuteContractResponse = {
+  txHash: string;
+};
+
+export type MigrateContractRequest = {
+  contractAddr: string;
+  newCodeId: number;
+  migrateMsg?: object;
+};
+export type MigrateContractResponse = {
+  txHash: string;
+};
+
+export type UpdateContractAdminRequest = {
+  newAdmin: string;
+  contractAddr: string;
+};
+export type UpdateContractAdminResponse = {
+  txHash: string;
+};
+
+// if a transaction fails, methods will raise an error
```

Comment:
I think it would be good to document this class more extensively. Explain what each method does, how to invoke them, and any special cases or concerns the caller needs to worry about. (you've done some of the last thing by documenting assumptions)

you probably don't have to document the types ^ as long as these methods are documented.